test/static.test.js

```diff
@@ -1086,3 +1115,473 @@ t.test('register with wildcard false', t => {
+      })
+      req.on('error', err => console.error(err))
+      req.end()
+
+      simple.concat({
+        method: 'GET',
+        url: 'http://localhost:' + fastify.server.address().port + '/static?a=b'
+      }, (err, response, body) => {
+        t.error(err)
+        t.strictEqual(response.statusCode, 200)
+        t.strictEqual(body.toString(), indexContent)
+        genericResponseChecks(t, response)
+      })
+    })
+
+    t.test('/static/', t => {
+      t.plan(3 + GENERIC_RESPONSE_CHECK_COUNT)
+
+      simple.concat({
+        method: 'GET',
+        url: 'http://localhost:' + fastify.server.address().port + '/static/'
+      }, (err, response, body) => {
+        t.error(err)
+        t.strictEqual(response.statusCode, 200)
+        t.strictEqual(body.toString(), indexContent)
+        genericResponseChecks(t, response)
+      })
+    })
+
+    t.test('/static/deep', t => {
+      t.plan(2 + GENERIC_ERROR_RESPONSE_CHECK_COUNT)
+
+      simple.concat({
+        method: 'GET',
+        url: 'http://localhost:' + fastify.server.address().port + '/static/deep'
+      }, (err, response, body) => {
+        t.error(err)
+        t.strictEqual(response.statusCode, 404)
+        genericErrorResponseChecks(t, response)
+      })
+    })
+
+    t.test('/static/deep/path/for/test?a=b', t => {
+      t.plan(5 + GENERIC_RESPONSE_CHECK_COUNT)
+
+      // simple-get doesn't tell us about redirects so use http.request directly
+      const testurl = 'http://localhost:' + fastify.server.address().port + '/static/deep/path/for/test?a=b'
+      const req = http.request(url.parse(testurl), res => {
+        t.strictEqual(res.statusCode, 301)
+        t.strictEqual(res.headers['location'], '/static/deep/path/for/test/?a=b')
+      })
+      req.on('error', err => console.error(err))
+      req.end()
+
+      // verify the redirect with query parameters works
+      simple.concat({
+        method: 'GET',
+        url: testurl
+      }, (err, response, body) => {
+        t.error(err)
+        t.strictEqual(response.statusCode, 200)
+        t.strictEqual(body.toString(), innerIndex)
+        genericResponseChecks(t, response)
+      })
+    })
+
+    t.test('/static/deep/path/for/test', t => {
+      t.plan(3 + GENERIC_RESPONSE_CHECK_COUNT)
+
+      simple.concat({
+        method: 'GET',
+        url: 'http://localhost:' + fastify.server.address().port + '/static/deep/path/for/test'
+      }, (err, response, body) => {
+        t.error(err)
+        t.strictEqual(response.statusCode, 200)
+        t.strictEqual(body.toString(), innerIndex)
+        genericResponseChecks(t, response)
+      })
+    })
+  })
+})
+
+t.test('register /static with redirect true and wildcard false', t => {
+  t.plan(6)
+
+  const pluginOptions = {
+    root: path.join(__dirname, '/static'),
+    prefix: '/static',
+    redirect: true,
+    wildcard: false,
+    index: 'index.html'
+  }
+
+  const fastify = Fastify()
```

Comment:
Can you add another test with `ignoreTrailingSlash: true`?

Reply:
I tested `redirect: true, wildcard: true` with `ignoreTrailingSlash: true` on the server and it worked as expected without any further changes.

As is `redirect: true, wildcard: false` cannot be used on a server with `ignoreTrailingSlash: true`.  To fix this would require a way to determine if `ignoreTrailingSlash` is enabled.  In my working copy I edited `node_modules/fastify/fastify.js` so that the server object is created with:
```js
  const fastify = {
    ignoreTrailingSlash: options.ignoreTrailingSlash,
    [kChildren]: []
  }
```

Then I made my local copy of `fastify-static` check if `fastify.ignoreTrailingSlash` is enabled to know how to register routes.  The problem is that `fastify.get('/static/')` creates two routes so `wildcard: false` needs to know to not also setup `fastify.get('/static')`, as it stands `find-my-way` throws an assertion.  I'd prefer to not use a `try / catch` block to suppress this assertion for risk of potentially suppressing errors.

An alternative would be for `fastify-static` to register `fastify.get(pathname.replace(/\/$/, ''))` before `fastify.get(pathname)`, and fix `find-my-way` to only register the `ignoreTrailingSlash` route if it doesn't already exist.  This would probably be cleaner for `fastify-static`.

Reply:
Can you please document the behavior that is not supported? We might want to implement this in a follow-up a PR if there is a need.

Reply:
Done.

I looked at `find-my-route` again, it appears that `/path` registers `/path/` and `/path/` registers `/path`.  So probably the best fix would be to honor `ignoreTrailingSlash: false` option for find-my-way routes to override the global option, that way we could pass `ignoreTrailingSlash: false` for `wildcard: false, redirect: true`.